test/k4FWCoreTest/src/components/ExampleFunctionalConsumerRuntimeCollections.cpp

```diff
@@ -26,20 +26,26 @@
 #include <string>
 
 struct ExampleFunctionalConsumerRuntimeCollections final
-    : k4FWCore::Consumer<void(const std::map<std::string, const edm4hep::MCParticleCollection&>& input)> {
+    : k4FWCore::Consumer<void(const std::vector<const edm4hep::MCParticleCollection*>& input)> {
   // The pair in KeyValue can be changed from python and it corresponds
   // to the name of the output collection
   ExampleFunctionalConsumerRuntimeCollections(const std::string& name, ISvcLocator* svcLoc)
       : Consumer(name, svcLoc, KeyValues("InputCollection", {"DefaultValue"})) {}
 
   // This is the function that will be called to produce the data
-  void operator()(const std::map<std::string, const edm4hep::MCParticleCollection&>& input) const override {
+  void operator()(const std::vector<const edm4hep::MCParticleCollection*>& input) const override {
     if (input.size() != 3) {
-      throw std::runtime_error("Wrong size of the input map, expected 3, got " + std::to_string(input.size()));
+      throw std::runtime_error("Wrong size of the input vector, expected 3, got " + std::to_string(input.size()));
     }
-    for (auto& [key, val] : input) {
+    for (int i = 0; i < 3; i++) {
+      if (inputLocations(0)[i] != "MCParticles" + std::to_string(i)) {
```

Comment:
There is no easy way to have `inputLocations` take the key from the `KeyValues` above, right? I.e. such that something like this would becom possible

```cpp
if (inputLocations("InputCollection")[i] == "MCParticles" + std::to_string(i)) { /**/ }
```

It doesn't really matter here, but if there are multiple input variable length inputs, this would make it much harder to break things, I think. However, I am not sure how often that actually is the case in real world uses.

Reply:
There is:
- either by property (returning `std::vector<std::string>>`)
  ```cpp
  auto inputLocations(const std::string& name) const {
    Gaudi::Property<std::vector<std::string>> tmp;
    tmp.assign(this->getProperty(name));
    return tmp.value();
  }
  ```
 - or from `m_inputlocations` (returing view)
  ```cpp
  auto intpuLocations(const std::string& name) const {
    auto it = std::ranges::find_if(m_inputLocations, [&name](const auto& prop) { return prop.name() == name; });
    if (it == m_inputLocations.end()) {
      throw GaudiException("Input named " + name + " not found", "Consumer", StatusCode::FAILURE);
    }
    return *it | std::views::transform([](const DataObjID& id) -> const auto& { return id.key(); });
  }
  ```

It could be also nice to get by type (if it's unique, there is a tuple underneath) but that seems to be more complicated as `detail::transformType` transforms all the collection types to `podio::CollectionBase`

Reply:
Added functions to get by name. Getting by type is not trivial since one still has to get the index to find where the vector with the names is... so maybe if someone asks for it? By index and name should be plenty of options.

Reply:
I think that the use case for getting this by type is limited(?). i would assume that in that case you don't really care about the collection name since it's only one in any case. The only reason I could think of would be for log messages, but in that case, one of the overloads here with index or name, should also work, right?

Reply:
Yes, I agree, it's complicated to implement and there are already ways to achieve the same result so I would bother with it now. I mentioned it mostly for completeness since they have it in Gaudi.

I could think of something like this: when there multiple arguments like here or more
```cpp
using FloatColl         = std::vector<const podio::UserDataCollection<float>*>;
using ParticleColl      = std::vector<const edm4hep::MCParticleCollection*>;
using SimTrackerHitColl = std::vector<const edm4hep::SimTrackerHitCollection*>;
using TrackerHitColl    = std::vector<const edm4hep::TrackerHit3DCollection*>;
using TrackColl         = std::vector<const edm4hep::TrackCollection*>;

using retType = std::tuple<std::vector<podio::UserDataCollection<float>>, std::vector<edm4hep::MCParticleCollection>,
                           std::vector<edm4hep::MCParticleCollection>, std::vector<edm4hep::SimTrackerHitCollection>,
                           std::vector<edm4hep::TrackerHit3DCollection>, std::vector<edm4hep::TrackCollection>>;

retType operator()(const FloatColl& floatVec, const ParticleColl& particlesVec,
                     const SimTrackerHitColl& simTrackerHitVec, const TrackerHitColl& trackerHitVec,
                     const TrackColl& trackVec) const
```
I think it'd be nice to avoid counting the arguments to write ~~`outputLocations(4)`~~ `outputLocations(5)` or scrolling to the constructor to find the name in `KeyValues`, and instead just write `outputLocations<std::vector<edm4hep::TrackCollection>>()` when I want to get the name for the locations of that "vector of track-collection" output

Reply:
I don't have a strong opinion about this, it can be useful. But so far I think we'll have few algorithms with any number of inputs or outputs so the reach will be small.